core/src/test/java/com/linecorp/armeria/internal/common/PathAndQueryTest.java

```diff
@@ -544,4 +544,16 @@ private static PathAndQuery parse(@Nullable String rawPath, boolean allowDoubleD
         }
         return res;
     }
+
+    @Test
+    void assertSquareBracketsInPath() {
+                final PathAndQuery res = parse("/#/:@[]!$&'()*+,;=");
+        assertThat(res).isNotNull();
+        assertThat(res.path()).isNotEqualTo("/#/:@!$&'()*+,;=");
```

Comment:
How about showing the path instead so that reviewers can easily check the returned value?
`assertThat(res.path()).isEqualTo("/#/:@%5B%5D!$&'()*+,;=");`